src/components/City.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { API_BASE_URL, API_KEY } from '../api';
+import { setCity } from '../redux/city/citySlice';
+
+const City = () => {
+  const dispatch = useDispatch();
+  const city = useSelector((state) => state.city);
+  const { state: selectedState, city: selectedCity } = useParams();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const response = await fetch(`${API_BASE_URL}/city?city=${selectedCity}&state=${selectedState}&country=Canada&key=${API_KEY}`);
+      if (!response.ok) {
+        throw new Error('Failed to fetch data');
+      }
+      const data = await response.json();
+      dispatch(setCity(data.data));
+      setLoading(false);
+    } catch (error) {
+      setError(error);
+      setLoading(false);
+    }
+  }, [dispatch, selectedState, selectedCity]);
+
+  useEffect(() => {
+    fetchData();
+  }, [fetchData, selectedState, selectedCity]);
+
+  const retryFetch = () => {
+    setError(null);
+    setLoading(true);
+    fetchData();
+  };
+
+  const renderContent = () => {
+    if (loading) {
+      return <p className="loading">Loading...</p>;
+    }
+    if (error) {
+      return (
+        <div className="error">
+          <p>Error fetching data. Please try again.</p>
+          <button type="button" onClick={retryFetch}>Retry</button>
+        </div>
+      );
+    }
+    const { current } = city;
+    return (
+      <div className="container">
+        <nav className="navbar flex">
+          <Link to={`/${selectedState}`} className="back">Back</Link>
```

Comment:
- [ ] As required for the project to follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)) including `Typography`(font face, size, and weight), `Layout`(composition and space between elements), I think it would be better if you improve this back button design by also add the missing `ShowLeftArrow`